docs/pages/kubernetes-access/discovery.mdx

```diff
@@ -0,0 +1,106 @@
+---
+title: Kubernetes Auto-Discovery (Preview)
+description: Detailed guides for configuring Kubernetes Auto-Discovery.
+layout: tocless-doc
+---
+
+Kubernetes Auto-Discovery is a Teleport feature that allows Kubernetes clusters 
+hosted on cloud providers to be discovered and enrolled automatically. 
+
+While discovering a new Kubernetes cluster, Teleport does not install any component 
+on the cluster. Instead, it requires direct access to the cluster's API and minimal permissions.
+
+## How Kubernetes Auto-Discovery works
+
+The Kubernetes Auto-Discovery consists of two distinct steps:
+
+### Polling cloud APIs
+
+The Teleport Discovery Service is responsible for scanning the configured cloud 
+providers and identifying if any Kubernetes cluster match the filtering labels. 
+When the process identifies a new Kubernetes cluster, it creates a dynamic 
+resource within Teleport. This resource includes information imported from the 
+cloud cluster such as:
+
+- *Name*: Cluster Name (use `teleport.dev/kubernetes-name` tag to import the cluster in Teleport with a custom name) 
```

Comment:
I moved this into a notice block. 
The tag is not set in the teleport resource. It must be set in the EKS or AKS cluster so that auto-discovery uses it as the name of the imported resource in order to prevent collisions.